launch_ros/launch_ros/event_handlers/on_state_transition.py

```diff
@@ -34,7 +33,7 @@ def __init__(
         self,
         *,
         entities: SomeEntitiesType,
-        target_lifecycle_node: LifecycleNode = None,
+        target_lifecycle_node: Optional[SomeSubstitutionsType] = None,
```

Comment:
Why does this need to be a `SomeSubstitutionsType`? Not saying it's wrong, just curious what use case you have in mind?

Reply:
And further, I'm not sure this works because this would essentially be a type name as a string, but before this was a _value_ not a _type_, and we passed that instance of the object in to the event as `event.action`. So maybe I do think this isn't right.

Reply:
Because I needed this to not be specifically a `LifecycleNode` due to circular dependencies using this within LifecycleNode itself for the transitioning. So, to make sure that it is still a lifecycle-compliant node, I used the Optional type and checked that it was a proper instance of a lifecycle-compliant node using `is_lifecycle_node` (since components, lifecycle, and node in launch_ros can all be used to launch a rclcpp::LifecycleNode)

> Im not sure this works 

I can't comment on the reasoning you have for it not working, but I can say from testing with the examples I provided that it does indeed work.

Reply:
> Isn't LifecycleNode a type and Optional[SomeSubstitutionsType] a type? 

I mean, yes, but that's like saying replacing `rclcpp::Node` with `std::string` in c++ is ok because they're both types...

If it's just an issue with circular dependencies then for the Type annotations you can just string quote the type, e.g.:

```suggestion
        target_lifecycle_node: Optional['LifecycleNode'] = None,
```

Or I think you can do something like this at the top of the file:

```
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from ... import LifecycleNode
```

See: https://stackoverflow.com/a/39757388

Reply:
btw, didn't mean to come off snarky with the c++ comparison, but that is the sort of what's happening with the `SomeSubstitutionsType`, since that type essentially gets resolved into a string.

Reply:
Hey, no worries. I'm not overly familiar with this in Python3 so point taken

Fixed in 4c06f44e78c05ace536932a759ed4882aa636481